aws/resource_aws_emr_instance_fleet_test.go

```diff
@@ -0,0 +1,588 @@
+}
+
+resource "aws_iam_role_policy_attachment" "profile-attach" {
+  role       = "${aws_iam_role.iam_emr_profile_role.id}"
+  policy_arn = "${aws_iam_policy.iam_emr_profile_policy.arn}"
+}
+
+resource "aws_iam_policy" "iam_emr_profile_policy" {
+  name = "iam_emr_profile_policy_%d"
+
+  policy = <<EOT
+{
+    "Version": "2012-10-17",
+    "Statement": [{
+        "Effect": "Allow",
+        "Resource": "*",
+        "Action": [
+            "cloudwatch:*",
+            "dynamodb:*",
+            "ec2:Describe*",
+            "elasticmapreduce:Describe*",
+            "elasticmapreduce:ListBootstrapActions",
+            "elasticmapreduce:ListClusters",
+            "elasticmapreduce:ListInstanceGroups",
+            "elasticmapreduce:ListInstances",
+            "elasticmapreduce:ListSteps",
+            "kinesis:CreateStream",
+            "kinesis:DeleteStream",
+            "kinesis:DescribeStream",
+            "kinesis:GetRecords",
+            "kinesis:GetShardIterator",
+            "kinesis:MergeShards",
+            "kinesis:PutRecord",
+            "kinesis:SplitShard",
+            "rds:Describe*",
+            "s3:*",
+            "sdb:*",
+            "sns:*",
+            "sqs:*"
+        ]
+    }]
+}
+EOT
+}
+`
+
+func testAccAWSEmrInstanceFleetConfig(r int) string {
+	return fmt.Sprintf(testAccAWSEmrInstanceFleetBase+`
+	resource "aws_emr_instance_fleet" "task" {
+      cluster_id            = "${aws_emr_cluster.tf-test-cluster.id}"
+	  instance_fleet_type   = "TASK"
+	  instance_type_configs = [
+		{
+          instance_type = "m3.xlarge"
+        }
+      ]
+      launch_specifications {
+	    spot_specification {
+		  timeout_action           = "TERMINATE_CLUSTER"
+          timeout_duration_minutes = 10
+	    }
+	  }
+	  name                      = "emr_instance_fleet_%d"
+	  target_on_demand_capacity = 0
+	  target_spot_capacity      = 1
+	}
+	`, r, r, r, r, r, r, r)
+}
+
+func testAccAWSEmrInstanceFleetConfigZeroCount(r int) string {
+	return fmt.Sprintf(testAccAWSEmrInstanceFleetBase+`
+	resource "aws_emr_instance_fleet" "task" {
+      cluster_id            = "${aws_emr_cluster.tf-test-cluster.id}"
+	  instance_fleet_type   = "TASK"
+	  instance_type_configs = [
+		{
+          instance_type = "m3.xlarge"
+        }
+      ]
+      launch_specifications {
+	    spot_specification {
+		  timeout_action           = "TERMINATE_CLUSTER"
+          timeout_duration_minutes = 10
+	    }
+	  }
+	  name                      = "emr_instance_fleet_%d"
+	  target_on_demand_capacity = 0
+	  target_spot_capacity      = 0
+	}
+	`, r, r, r, r, r, r, r)
+}
+
+func testAccAWSEmrInstanceFleetConfigEbsBasic(r int) string {
+	return fmt.Sprintf(testAccAWSEmrInstanceFleetBase+`
+	resource "aws_emr_instance_fleet" "task" {
+	  cluster_id            = "${aws_emr_cluster.tf-test-cluster.id}"
+	  instance_fleet_type   = "TASK"
+	  instance_type_configs = [
+		{
+          ebs_optimized = true
+		  ebs_config = [
+			{
+              size = 10
```

Comment:
The formatting of the resources seems to be muddled in some of the tests. It would be nice to fix this to improve readability.